doc/mpls/MPLS_hld.md

```diff
@@ -0,0 +1,510 @@
+          - [Functions](#functions-2)
+        - [RouteOrch](#routeorch)
+          - [Functions](#functions-3)
+        - [NeighOrch](#neighorch)
+          - [Functions](#functions-4)
+        - [CrmOrch](#crmorch)
+          - [Functions](#functions-4)
+        - [Label/LabelStack](#labellabelstack)
+        - [NextHopKey](#nexthopkey)
+        - [Syncd](#syncd)
+    - [SAI API](#sai-api)
+      - [Router Interface](#router-interface)
+      - [MPLS](#mpls)
+      - [Next Hop](#next-hop)
+    - [Configuration and management](#configuration-and-management)
+      - [CLI Enhancements](#cli-enhancements)
+      - [Config DB Enhancements](#config-db-enhancements)
+        - [INTERFACE](#interface)
+        - [PORTCHANNEL_INTERFACE](#portchannel_interface)
+    - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+    - [Restrictions/Limitations](#restrictionslimitations)
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+      - [System Test cases](#system-test-cases)
+    - [Open/Action items - if any](#openaction-items---if-any)
+### Revision
+|  Rev  |  Date       |  Author  | Change Description |
+| :---: | :---------: | :------: | ------------------ |
+|  0.1  | Jan-10-2021 | A Pokora | Initial version    |
+|  0.2  | Jan-19-2021 | A Pokora | Updates from MPLS community review |
+
+### Scope
+
+This document provides general information about the MPLS feature implementation in SONiC.
+
+### Definitions/Abbreviations
+| Abbreviation | Description                           |
+| ------------ | ------------------------------------- |
+| cRPD         | Containerized Routing Protocol Daemon |
+| MPLS         | Multi-Label Packet Switching          |
+
+### Overview
+
+This document provides general information about the MPLS feature implementation in SONiC.
+
+### Requirements
+
+This section describes the SONiC requirements for the MPLS feature.
+
+#### Functional requirements
+- Support to enable/disable MPLS per Interface.
+- Support for MPLS Push, Pop, and Swap routes.
+- Support for bulk MPLS in-segment entry SAI programming.
+- Support for CRM monitoring of MPLS in-segment used/available entries.
+
+#### Configuration and Management requirements
+- SONiC CLI support for MPLS enable/disable per Router Interface.
+- Configurable CRM thresholds for monitoring MPLS in-segment entries used/available.
+
+#### Scalability Requirements
+- Up to max ASIC capable MPLS routes are supported.
+- Error is logged in syslog for all attempted MPLS routes after max limit is reached.
+- CRM notification upon reaching configurable scaling thresholds.
+
+#### Warm Boot Requirements
+- MPLS functionality continues across warm reboot.
+- Support for planned system warm restart.
+- Support for SWSS docker warm restart.
+
+#### Future Requirements
+- Support for VRFs
+- SONiC CLI support for operational commands.
+- Integration with FRR routing stack.
+- Integration with VS SAI.
+
```

Comment:
Looks like this HLD covers only  PHY & port-channel router interfaces as MPLS capable interfaces, not other interfaces e.g VLAN interface, please mention it here if we have a plan to add support for it in the future.

Reply:
The MPLS enable is on router interface attribute in this HLD.